src/Core/src/Platform/Windows/ButtonExtensions.cs

```diff
@@ -171,9 +171,13 @@ void OnImageOpened(object sender, RoutedEventArgs e)
 							{
 								// do the actual resize
 								var imageSourceSize = actualImageSource.GetImageSourceSize(platformButton);
-								nativeImage.Width = imageSourceSize.Width;
-								nativeImage.Height = imageSourceSize.Height;
+
+								nativeImage.Width = button.Width > 0 ? platformButton.ActualWidth : imageSourceSize.Width;
```

Comment:
If ImageButton HeightRequest and WidthRequest properties have been set we use that values on the image. Otherwise, the height and width of the original image size is used.

Reply:
Applied changes to use `Primitives.Dimension.Unset`, also added a sample:
![fix-imagebutton-size](https://user-images.githubusercontent.com/6755973/167612649-c870815c-5035-4866-8351-0b4c318204f5.gif)

Is working correctly and also resize as expected.